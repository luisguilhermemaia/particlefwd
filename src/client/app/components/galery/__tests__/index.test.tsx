import '@testing-library/jest-dom/extend-expect';
import { RenderResult, render } from '@testing-library/react';
import React from 'react';

import { Gallery } from '..';

describe('Gallery', () => {
  let renderResult: RenderResult;
  const images = [
    {
      uri: 'https://particleforward.com/api/challenge/assets/image1',
      set: 'fe4cfedffdffffffff',
    },
    {
      uri: 'https://particleforward.com/api/challenge/assets/image1',
      set: 'fe4cfedffdffffffff',
    },
    {
      uri: 'https://particleforward.com/api/challenge/assets/image2',
      set: 'fe4cfedffdffffffff',
    },
    {
      uri: 'https://particleforward.com/api/challenge/assets/image3',
      set: 'fe4cfedffdffffffff',
    },
    {
      uri: 'https://particleforward.com/api/challenge/assets/image4',
      set: 'fe4cfedffdffffffff',
    },
  ];
  const doRender = () => {
    renderResult = render(<Gallery images={images} loading={false} />);
  };

  it('renders correctly with Gallery component', () => {
    doRender();

    const { container } = renderResult;

    expect(container).toMatchSnapshot();
  });
});
