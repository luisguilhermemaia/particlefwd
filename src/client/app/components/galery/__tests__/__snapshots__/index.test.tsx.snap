// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders correctly with Gallery component 1`] = `
<div>
  <div
    class="galery__StyledContainer-sc-1x9cw2l-0 dyJuLe"
  >
    <div
      class="galery__StyledGrid-sc-1x9cw2l-1 kHohyy"
    >
      <div
        class="galery__StyledDiv-sc-1x9cw2l-2 sRWto"
      >
        <img
          src="https://particleforward.com/api/challenge/assets/image1_2.jpg"
        />
      </div>
      <div
        class="galery__StyledDiv-sc-1x9cw2l-2 sRWto"
      >
        <img
          src="https://particleforward.com/api/challenge/assets/image1_2.jpg"
        />
      </div>
      <div
        class="galery__StyledDiv-sc-1x9cw2l-2 sRWto"
      >
        <img
          src="https://particleforward.com/api/challenge/assets/image2_2.jpg"
        />
      </div>
      <div
        class="galery__StyledDiv-sc-1x9cw2l-2 sRWto"
      >
        <img
          src="https://particleforward.com/api/challenge/assets/image3_2.jpg"
        />
      </div>
      <div
        class="galery__StyledDiv-sc-1x9cw2l-2 sRWto"
      >
        <img
          src="https://particleforward.com/api/challenge/assets/image4_2.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
